File: sdks/dart/lib/src/server/regions.dart
import '../server_client.dart';

class Regions {
  final ApplAdServer _client;

  Regions(this._client);

  Future<Map<String, dynamic>> list() async {
    return _client.get('/v1/regions');
  }

  Future<Map<String, dynamic>> get(String regionId) async {
    return _client.get('/v1/regions/$regionId');
  }

  Future<Map<String, dynamic>> getActive() async {
    return _client.get('/v1/regions/active');
  }

  Future<Map<String, dynamic>> setActive(String regionId) async {
    return _client.put('/v1/regions/active', data: {
      'regionId': regionId,
    });
  }

  Future<Map<String, dynamic>> getHealth(String regionId) async {
    return _client.get('/v1/regions/$regionId/health');
  }
}

File: sdks/python/applad/regions.py
"""Regions service."""


class Regions:
    def __init__(self, client):
        self.client = client

    def list(self):
        return self.client._call("GET", "/regions")

    def get(self, region_id: str):
        return self.client._call("GET", f"/regions/{region_id}")

    def get_active(self):
        return self.client._call("GET", "/regions/active")

    def set_active(self, region_id: str):
        return self.client._call("PUT", "/regions/active", {"regionId": region_id})

    def get_health(self, region_id: str):
        return self.client._call("GET", f"/regions/{region_id}/health")